### ISSUE TITLE
GigaChannelManager keeps checking the same torrent

### ISSUE THREAD

**Issue by qstokkink** [Tribler/tribler#4620]
My GigaChannelManager keeps updating the same torrent (log actually keeps giving me these messages, only bothered copy-pasting 2 of them):

```python
INFO    1561466834.21       gigachannel_manager:128  (GigaChannelManager)  Processing previously downloaded, but unprocessed channel torrent 02ef6751e809bc49c92540962fd8fa54cf89b0004c61bfa1d31b5095ee20127a0bda166852057e15cdf57e3c315b88e67fddaa4d4e41b34b9415163d6062ff00 ver 1560889103430->1560889103502
INFO    1561466839.27       gigachannel_manager:128  (GigaChannelManager)  Processing previously downloaded, but unprocessed channel torrent 02ef6751e809bc49c92540962fd8fa54cf89b0004c61bfa1d31b5095ee20127a0bda166852057e15cdf57e3c315b88e67fddaa4d4e41b34b9415163d6062ff00 ver 1560889103430->1560889103502
```

And I should add: the particular channel is not actually progressing.

**Comment by qstokkink** [comment]
From some logging I see that this is only called once:

https://github.com/Tribler/tribler/blob/50607100bdc19cf5aac452626ec023bef15d22a7/Tribler/Core/Modules/gigachannel_manager.py#L205

**Comment by qstokkink** [comment]
Actually the `process_channel_dir` only being called once seems by design.

By applying this fix I got the channel to update again (albeit VERY SLOW):

```patch
diff --git a/Tribler/Core/Modules/gigachannel_manager.py b/Tribler/Core/Modules/gigachannel_manager.py
index 6658b832e..443c811c2 100644
--- a/Tribler/Core/Modules/gigachannel_manager.py
+++ b/Tribler/Core/Modules/gigachannel_manager.py
@@ -111,6 +111,9 @@ class GigaChannelManager(TaskManager):
         Check whether there are channels that are updated. If so, download the new version of the channel.
         """
         # FIXME: These naughty try-except-pass workarounds are necessary to keep the loop going in all circumstances
+        if self.processing:
+            self._logger.info("Skipping channel check, already busy.")
+            return
 
         with db_session:
             channels = list(self.session.lm.mds.ChannelMetadata.get_updated_channels())
```

**Comment by ichorid** [comment]
@qstokkink , you're right, the channel can be added to the queue several times, and that is definitely an error! I'll fix this and add the fix to #4595

**Comment by qstokkink** [comment]
@ichorid thanks 👍
